product_contract/models/sale_order.py

```diff
@@ -103,7 +103,9 @@ def action_confirm(self):
     @api.depends("order_line")
     def _compute_contract_count(self):
         for rec in self:
-            rec.contract_count = len(rec.order_line.mapped('contract_id'))
+            rec.contract_count = len(
```

Comment:
Look this to improve performance:
https://github.com/OCA/stock-logistics-workflow/blob/bb8c23a1044198e3b5a9b9c6edd08bfabf1786db/stock_picking_batch_extended/models/stock_batch_picking.py#L138-L147

Reply:
You can create a get_domain method to extend easily and keep the same logic in this module.

Reply:
Or You can use a config parameter to count only active contracts

Reply:
Database parameter (same for all companies):
https://github.com/odoo/odoo/blob/18b7cc85e23612970fbebb7bb7b0ac4bc00a7bad/addons/sale/models/res_config_settings.py#L139

Company parameter:
https://github.com/odoo/odoo/blob/18b7cc85e23612970fbebb7bb7b0ac4bc00a7bad/addons/sale/models/res_config_settings.py#L10

Reply:
Thanks a lot, I'm going to try.